moderngl_window/loaders/program/separate.py

```diff
@@ -10,29 +12,48 @@
 class Loader(BaseLoader):
     kind = 'separate'
 
-    def load(self) -> moderngl.Program:
+    def load(self) -> Union[
+        moderngl.Program,
+        moderngl.ComputeShader,
+        program.ComputeProgram,
+        program.ReloadableProgram
+    ]:
         """Loads a shader program were each shader is a separate file.
 
         This detected and dictated by the ``kind`` in the ``ProgramDescription``.
 
         Returns:
             moderngl.Program: The Program instance
         """
+        prog = None
+        compute_shader = None
+
         vs_source = self.load_shader("vertex", self.meta.vertex_shader)
         geo_source = self.load_shader("geometry", self.meta.geometry_shader)
         fs_source = self.load_shader("fragment", self.meta.fragment_shader)
         tc_source = self.load_shader("tess_control", self.meta.tess_control_shader)
         te_source = self.load_shader("tess_evaluation", self.meta.tess_evaluation_shader)
+        cs_source = self.load_shader("compute", self.meta.compute_shader)
+
+        if vs_source:
+            shaders = program.ProgramShaders.from_separate(
+                self.meta,
+                vs_source,
+                geometry_source=geo_source,
+                fragment_source=fs_source,
+                tess_control_source=tc_source,
+                tess_evaluation_source=te_source,
+            )
+            prog = shaders.create()

```

Comment:
Had to make it conditional so that you load a standalone compute shader

Reply:
`if `vs_source` elif `cs_source` else raise IncorrectlyConfigured` would be enough.